### ISSUE TITLE
Implement proper pagination

### ISSUE THREAD

**Issue by FredrikAugust** [Kodeworks/liquidator-frontend#124]
Right now, pagination isn't actually handled - it just returns the results from the fisrt page.

Naturally, this should be implemented before going into prod. I plan to implement it either as a generator function, or by keeping state in the store (latter tends to work better with react hooks).

**Comment by FredrikAugust** [comment]
I think I will solve it like this;

```typescript
// reducer
async function* doGetAllTransactions() {
  ...
}

// component
const getMoreTransactions = await doGetAllTransactions();

const [resourceExhausted, setResourceExhausted] = React.useState(getMoreTransactions.done);

const onLoadMoreClick = async () => {
  if (!resourceExhausted) {
    await getMoreTransactions.next();
  }
}
```

**Comment by FredrikAugust** [comment]
Alright, in an attempt to increase overall understanding of the complex parts of the application, I will write a short writeup of this feature here.

----------------------------------------------------

# Pagination

Pagination (as implemented by the backend) works like this;

- You request a set of data, e.g. `getAllTransactions`, which can consist of a theoretically infinite set of data
- The server will return a paginated response, which looks like this;
```json
{
  prev: "url which you can use to request the previous page of items",
  next: "url which you can use to request the next page of items",
  results: [] // this contains the actual results from this request
}
```
- When first requesting all transactions, upon entering the transaction page, you will be given one page of items, and to request more, you have to send a request to the "next" url

## Problem

As of now, there is no way to request the next page of results, meaning you are stuck with the initial results from the first page.

## Solution

### Prerequisites

- A solid understanding of javascript generators
  - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators
  - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*
- A good understanding of javascript asynchronous functions
- A relatively good understanding of how we store transactions in the store in this application, and how they end up there after a request. For this you also need to understand reducers/stores
  - https://www.robinwieruch.de/javascript-reducer/

### Implementation

The planned implementation will look like this;

When you use an action similar to `doGetAllTransactions`, you will fetch the first page of items, add them to store, and then be left with an async function generator. You should know how these work.

When you then want to request more items, you can simply call `iterator.next()`, and that will take care of fetching the next page, and adding it to the store, leading to your interface being updated auto-magically. To find out of you've fetched all the items, you simply check if the attribute `.done` of the last `.next()` call is `true`, and if so, just disable the `Fetch More` button, or whatever seems fitting in your situation.

**Comment by FredrikAugust** [comment]
Fixed by 8dbfe66ae13332d93ad01f8e16806ef6d070bbc5
